src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,133 @@
+
+        } else if (words[0].equals("delete")) {
+            if (words.length != 2) {
+                throw new DukeEventException(
+                        "OOPS!!! Please enter in format: delete <task_number>\n " +
+                                "e.g. delete 1");
+            } else {
+                try {
+                    int ranking = Integer.parseInt(words[1]);
+                    return new DeleteCommand(ranking);
+                } catch (Exception e) {
+                    throw new DukeException("Ranking should be a number! Call 'list' to see what tasks you have.");
+                }
+            }
+
+
+        } else if (words[0].equals("todo")) {
+            if (words.length != 2) {
+                throw new DukeEventException(
+                        "OOPS!!! Please enter in format: todo <task>\n " +
+                                "e.g. todo sing a song");
+            } else {
+                Task task = new ToDo(words[1]);
+                if (task == null) {
+                    throw new DukeDeadlineException("Todo not added...");
+                } else {
+                    return new AddCommand(task);
+                }
+            }
+
+        } else if (words[0].equals("event")) {
+            if (words.length != 2) {
+                throw new DukeEventException(
+                        "OOPS!!! Please enter in format: event <event> /at <event venue> \n " +
+                                "e.g. event Lesson /at Com1");
+            } else {
+                Task task = Event.setEvent(words[1]);
+                if (task == null) {
+                    throw new DukeDeadlineException("Event not added...");
+                } else {
+                    return new AddCommand(task);
+                }
+            }
+
+        } else if (words[0].equals("deadline")) {
+            if (words.length != 2) {
+                throw new DukeDeadlineException("OOPS!!! Please enter in format: deadline <task> /by <yyyy-mm-dd> \n " +
+                        "e.g. deadline complete project /by 2022-12-24");
+            } else {
+                Task task = Deadline.setDeadline(words[1]);
+                if (task == null) {
+                    throw new DukeDeadlineException("Deadline not added...");
+                } else {
+                    return new AddCommand(task);
+                }
+            }
+
+        } else if (words[0].equals("delete")) {
+            int ranking = Integer.parseInt(words[1]);
+            return new DeleteCommand(ranking);
+
+        } else if (words[0].equals("find")) {
+            try {
+//                LocalDate localdate = LocalDate.parse(words[1]);
+                return new FindCommand(words[1]);
+            } catch (Exception e) {
+                throw new DukeException("I'm not sure what's happening. Please try again later!");
+            }
+
+        } else {
+            throw new DukeException("Invalid Command!! Try something else hehe:)");
+        }
+    }
```

Comment:
I think we should avoid deep nesting here (especially since there is up to 3 levels of if-else statements in line 110).
Perhaps switch statements can be used to compare/parse the first word? :thinking: